File: code_samples/swift/utilities_string_manipulation.swift
import Foundation

/// Utilities: String Manipulation
/// AI/ML Training Sample

class StringManipulation {
    var data: String
    
    init() {
        self.data = ""
    }
    
    func process(_ input: String) {
        self.data = input
    }
    
    func getData() -> String {
        return self.data
    }
    
    func validate() -> Bool {
        return !self.data.isEmpty
    }
}

// Example usage
let instance = StringManipulation()
instance.process("example")
print("Data: \(instance.getData())")
print("Valid: \(instance.validate())")

File: code_samples/dart/utilities_string_manipulation.dart
// Utilities: String Manipulation
// AI/ML Training Sample

class StringManipulation {
  String _data = '';
  
  void process(String input) {
    _data = input;
  }
  
  String getData() => _data;
  
  bool validate() => _data.isNotEmpty;
}

void main() {
  final instance = StringManipulation();
  instance.process('example');
  print('Data: ${instance.getData()}');
  print('Valid: ${instance.validate()}');
}
